app/common/directives/language-switch.directive.js

```diff
@@ -17,11 +17,17 @@ function LanguageSwitchController($scope, Languages, TranslationService, $locati
     $scope.changeLanguage = changeLanguage;
     $scope.$on('event:authentication:login:succeeded', TranslationService.setStartLanguage);
     $scope.$on('event:authentication:logout:succeeded', TranslationService.setStartLanguage);
+    $scope.site = {};
```

Comment:
`$scope.site` is sent from the directive using the language-switch (see the parameter here: https://github.com/ushahidi/platform-client/blob/develop/app/common/directives/language-switch.directive.js#L9). You don't need to define it here since it overrides the information sent into the directive. This may be a bit confusing but the main reason is that we often have this information already in the place where we are using the directive, and also when using a directive in many different places, we want to have the flexibility to choose what information to send to the directive as a starting value. That is not possible if we define it again inside of the directive.

Instead, pass the values we need as a parameter from the mode-context-form-filter. We fetch the language here: https://github.com/ushahidi/platform-client/blob/develop/app/main/posts/views/mode-context-form-filter.directive.js#L94 and the variable holding the information we want is called `$scope.userLanguage`. Therefore, we can pass this information to the language-switch like this:
`<language-switch site="{language: userLanguage}"></language-switch>`